### ISSUE TITLE
ScrollCaret method.

### ISSUE THREAD

**Issue by VisitorQ** [jacobslusser/ScintillaNET#152]
Hello,

sorry to bother you again.
I've tried the "ScrollCaret" method and have noticed that it does not just scrolls the current position into view, it also selects the text from the position clicked to the current position newly set.
Here is a screenshot :
![scrollcaret](https://cloud.githubusercontent.com/assets/15521119/11559489/9bcc3406-99b9-11e5-9536-f578d2de06a5.png)

It is a nice feature by the way and one can of course use the "GotoPosition" method or the method "ClearSelections()" to remove the selection. There may be an opportunity to have two methods: "ScrollCaret" and "ScrollCaretWithSelection"...

All the best.

**Comment by jacobslusser** [comment]
See issue #128 for a brief explanation of the `ScrollRange` method.I think that's what you're looking for.

**Comment by VisitorQ** [comment]
I think I did not well explain my point. So, let's say that I open the context menu (see picture above) by a right-click at the position 650. I click the option "Go to Top", the property "CurrentPosition" is set to 0 then the "ScrollCaret" method is called. This method indeed moves, as expected, the caret to the position 0 and scrolls the viewport of the editor to the new position, 0, into view. BUT, it does not just do that, it ALSO selects the text from position 0 to 650 what is not expected. The "bug" (is this a bug?) is the subsequent selection of the text corresponding to the move of the caret from position 650 to position 0.

The method "ScrollRange" does not move the caret as far as I understand it, only the editor's viewport...

**Comment by jacobslusser** [comment]
I think what's missing from your equation is the `AnchorPosition`. The `AnchorPosition` and the `CurrentPostion` work together to define the range of a selection. You're moving one but not the other--thus adjusting your selection range. Try setting:

```cs
scintilla.AnchorPosition = scintilla.CurrentPosition = 0;
scintilla.ScrollCaret();
```

Also take a look at the `Scintilla.SetSel` and `Scintilla.SetSelection` methods.

**Comment by VisitorQ** [comment]
OK, I see... I did not know this property: "AnchorPosition". I've checked the Scintilla website and now 
I understand this behaviour. On the Scintilla site, I read :
"Scintilla maintains a selection that stretches between two points, the anchor and the current position. If the anchor and the current position are the same, there is no selected text."

So, everything works properly! Issue closed, it just emphasizes my lack of "know how"...

All the best.
